publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java

```diff
@@ -119,14 +120,21 @@ public class ResourceService extends ServiceWithTransactions {
     private final CounterService counterService;
     private final CustomerService customerService;
 
-    protected ResourceService(AmazonDynamoDB dynamoDBClient, String tableName, Clock clock,
-                              Supplier<SortableIdentifier> identifierSupplier, RawContentRetriever uriRetriever,
-                              ChannelClaimClient channelClaimClient,
-                              CustomerService customerService) {
+    public ResourceService(AmazonDynamoDB dynamoDBClient,
+                           String tableName, Clock clock,
+                           RawContentRetriever uriRetriever,
+                           ChannelClaimClient channelClaimClient,
+                           CustomerService customerService) {
         super(dynamoDBClient);
+
+        requireNonNull(dynamoDBClient, "DynamoDbClient cannot be null");
+        requireNonNull(tableName, "Table name cannot be null");
+        requireNonNull(uriRetriever, "UriRetriever name cannot be null");
+        requireNonNull(channelClaimClient, "ChannelClaimClient name cannot be null");
+        requireNonNull(customerService, "CustomerService name cannot be null");
+
```

Comment:
Not sure this is the best way of validating service on its instantiation, probably better with factory method, but I think it is OK, cause validation rules are simple :)